### PR TITLE
Add LabelFilterer and Store wrapper

### PR DESCRIPTION
**Pull request by MichelHollands**
Signed-off-by: Michel Hollands <michel.hollands@grafana.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

**Comment by cyriltovena** [comment]
I have a feeling that you actually don't need all of this and you can just call the querier label method on your side and filter them there by re-implementing the handler.

Same goes for the storage wrapper since you pass a storage you could wrap it before passing it down to this function. ?

File: pkg/ingester/ingester.go [hunk 1]
```diff
@@ -173,6 +174,7 @@ type Ingester struct {
 	wal WAL
 
 	chunkFilter storage.RequestChunkFilterer
+	labelFilter listutil.LabelValueFilterer
```
Comment (cyriltovena) [inline pkg/ingester/ingester.go:177]: This `LabelValueFilterer` interface is only used here, so I'd like that we defined it here only.

File: pkg/ingester/ingester.go [hunk 2]
```diff
@@ -717,8 +727,25 @@ func (i *Ingester) Label(ctx context.Context, req *logproto.LabelRequest) (*logp
 		}
 	}
 
+	allValues := listutil.MergeStringLists(resp.Values, storeValues)
+
+	if req.Values {
```
Comment (cyriltovena) [inline pkg/ingester/ingester.go:732]: Can you write a test for this, I think there's a bug when there's `Values` and no filter.

**Comment by cyriltovena** [comment]
> I have a feeling that you actually don't need all of this and you can just call the querier label method on your side and filter them there by re-implementing the handler.
> 
> Same goes for the storage wrapper since you pass a storage you could wrap it before passing it down to this function. ?

We discussed this offline and it seems more complex than I thought. Let's keep it this way and we can revisit if we have a better idea.

File: pkg/ingester/ingester.go (hunk 1 above)
Comment (MichelHollands) [reply pkg/ingester/ingester.go]: Moved.

File: pkg/ingester/ingester.go (hunk 2 above)
Comment (MichelHollands) [reply pkg/ingester/ingester.go]: Indeed there was a bug there. Tests were added.

**Review by cyriltovena — APPROVED** [review]
LGTM

**Pull request merged**